### PR TITLE
Testing whether PR ENV vars only appear from cross-repo PRs

### PR DESCRIPTION
**Pull request by orta**
This is a clone of #505 - my PRs were being classed as branch based, instead of  PR based CI builds.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please add a changelog entry for your changes.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>These new JS files do not have Flow enabled: dangerfile.js</td>
    </tr>
  </tbody>
</table>


If you are new to Flow, here are some resources: 

 * [Main Site](https://flowtype.org)
 * [Types](https://flowtype.org/docs/quick-reference.html#primitives)
 * [What is Flow?](https://code.facebook.com/posts/1505962329687926/flow-a-new-static-type-checker-for-javascript/)
 * [Danger's flow glossary](https://github.com/danger/danger-js/blob/master/docs/js_glossary.md)
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
OK 👍 this is good for review

File: dangerfile.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+// Request a CHANGELOG entry
+const hasChangelog = changedFiles.includes('changelog.md');
+if (!hasChangelog) { fail('Please add a changelog entry for your changedFiles.'); }
+
+// Politely ask for their name on the entry too
+const changelogDiff = danger.git.diffForFile('changelog.md');
+const contributorName = danger.github.pr.user.login;
+if (changelogDiff && changelogDiff.indexOf(contributorName) === -1) {
+  warn('Please add your GitHub name to "' + contributorName + '" the changelog entry.');
```
Comment (alloy) [inline dangerfile.js]: This message doesn’t seem right, should the “to” move after the contributor name?

**Comment by orta** [comment]
Shipped

**Comment by orta** [comment]
Added a check for assignees too - so it should fail now

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please set a reviewer</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by alloy** [comment]
🌱

**Pull request merged**